### DISCUSSION TITLE
TreeTable : Automatically change selection on arrow up/down

### DISCUSSION THREAD

**Discussion by ab-sara** [primefaces/community#458] [PrimeReact]
Hello everyone,

I'm currently working on a **TreeTable**, and I'm looking to implement an automatic selection change when the **arrow up/down keys** are pressed. The challenge I'm facing is that the event triggered by **onKeyDown** doesn't provide the target row data or the key necessary for setting the **selectionKeys**.

Does anyone have a solution or workaround for this issue?

Thank you!

**Comment by mertsincan** [comment]
Sorry for the delayed response! @gucal @nitrogenous can you assist with this question?

**Comment by nitrogenous** [comment]
Hey @ab-sara I have understand your point and I have faced with the same issue as well. I created an issue about this and I hope we will fix this problem in the next release. 

https://github.com/primefaces/primereact/issues/5815

**Reply by ab-sara** [reply]
Thank you so much , i really need that
